### PR TITLE
drop python 3.0-3.2 support code from setup.py

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
addresses #1627

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -27,10 +27,12 @@
 
   Thanks to `@RedBeardCode`_ for the PR(`#1664`_)
 
+* removed support code for python 3 < 3.3 addresssing (`#1627`_)
```
Comment (The-Compiler) [inline CHANGELOG.rst]: s/sss/ss/ (hah!)

Comment (nicoddemus) [inline CHANGELOG.rst]: :sweat_smile:

**Comment by nicoddemus** [comment]
LGTM, feel free to merge after fixing the small typo. :grin:

File: setup.py [hunk 2]
```diff
@@ -51,10 +51,10 @@ def main():
     install_requires = ['py>=1.4.29']  # pluggy is vendored in _pytest.vendored_packages
     extras_require = {}
     if has_environment_marker_support():
-        extras_require[':python_version=="2.6" or python_version=="3.0" or python_version=="3.1"'] = ['argparse']
+        extras_require[':python_version=="2.6"'] = ['argparse']
         extras_require[':sys_platform=="win32"'] = ['colorama']
     else:
-        if sys.version_info < (2, 7) or (3,) <= sys.version_info < (3, 2):
+        if sys.version_info < (2, 7):
```
Comment (nicoddemus) [inline setup.py]: This might be as well `sys.version_info[:2] == (2, 6)`, for consistency with the marker usage in `54`

**Comment by The-Compiler** [comment]
There is more stuff which should be changed/dropped to properly remove support

**Comment by RonnyPfannschmidt** [comment]
@The-Compiler in this pr i just want to kill the documented/normal install one

i plan to add some more PR's to remove strange code

for example i stumbled on a python 2.4 support thing ^^

**Comment by The-Compiler** [comment]
- pytest.org still says "runs on Posix/Windows, Python 2.6-3.5"
- `_pytest/assertion/rewrite.py` has various 3.0/3.1/3.2 support code (search for `REWRITE_NEWLINES` and `3.2+`)
- `README.rst` says `3.2+`
- `testing/code/test_source.py` has an xfail for `< (3,2)` which can be removed

**Comment by The-Compiler** [comment]
Okay, but at least adjust the documentation then.

**Comment by RonnyPfannschmidt** [comment]
im on it

**Comment by nicoddemus** [comment]
Oh sorry, I didn't have the code with me to check the docs... agree that we should properly drop the support in a single PR (setup.py, docs, code, etc)

**Comment by RonnyPfannschmidt** [comment]
should i drop anything else? (atm i really only want to drop docs and mentions, then phase bits of support code out in the `3.x` series)

**Comment by The-Compiler** [comment]
I'm with @nicoddemus - not sure what the rationale for only dropping half of it is? If we drop the code in this PR as well, we can be sure it's actually gone and we don't forget about it :wink:

**Comment by RonnyPfannschmidt** [comment]
ok

**Comment by RonnyPfannschmidt** [comment]
after investigating the support code for getstatementrange, it addresses python 2.4 and should be removed in a different pr

as for the assertion rewriter, - the exact code are conditionals also valid for python 2.6, and i would like to leave them until python 2.6 removal

**Comment by The-Compiler** [comment]
> as for the assertion rewriter, - the exact code are conditionals also valid for python 2.6, and i would like to leave them until python 2.6 removal

Oh, right, didn't think of that.

Could you still adjust the xfail in `test_source` though?

```python
@pytest.mark.xfail("sys.version_info[:3] < (2,7,0) or "
    "((3,0) <= sys.version_info[:2] < (3,2))")
def test_source_of_class_at_eof_without_newline(tmpdir):
```

**Comment by RonnyPfannschmidt** [comment]
i think i would like to remove that as we remove python2.6 as well, since we right now still have a testsuite that can run on 3.2, does any enterprise distro keep users on that?

**Comment by The-Compiler** [comment]
Huh - what's the point of having a testsuite passing on 3.2 when we say we drop 3.2 support? :wink:

As for conservative distros, I'm pretty sure they'll either stick with Python 2 altogether, or not going to use pytest 3.0. Hell, pip dropped it :wink:

**Comment by RonnyPfannschmidt** [comment]
true, will change the condition :)

**Comment by The-Compiler** [comment]
I'd like to see appveyor pass here, but it doesn't, see #1690.

@nicoddemus to the rescue? Or anyone else on Windows? :laughing:

**Comment by nicoddemus** [comment]
:grimacing:

**Comment by The-Compiler** [comment]
Thanks! :sparkles:

**Pull request merged**